crates/services/sync/src/ports.rs

```diff
@@ -12,10 +12,33 @@ use fuel_core_types::{
     },
     fuel_tx::Transaction,
     fuel_types::BlockHeight,
-    services::p2p::SourcePeer,
+    services::p2p::{
+        PeerId,
+        SourcePeer,
+    },
 };
 use std::ops::Range;
 
+/// Possible reasons to report a peer
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum PeerReportReason {
+    // Good
+    /// Successfully imported block
+    SuccessfulBlockImport,
+
+    // Bad
+    /// Did not receive advertised block headers
+    MissingBlockHeaders,
+    /// Report a peer for sending a bad block header
+    BadBlockHeader,
+    /// Did not receive advertised transactions
+    MissingTransactions,
+    /// Received invalid transactions
+    InvalidTransactions,
+    /// Received invalid block
+    InvalidBlock,
```

Comment:
If a block that passes all prior checks such as the PoA signature fails to import, we may want to consider this as a more serious case that indicates a fork or out of date client. Maybe the node should just stop syncing altogether if it hits this point rather than downvoting the main PoA node?

This could be a follow up as it likely requires more research.

Reply:
Interesting. I thought the previous checks were to show that the merkle root matches the txs, not that the txs are valid for the current state of the chain. Sounds like I'm missing something.

Reply:
Oh. But we've already checked consensus, so that's more important than checking individual txs at this level. Okay. I can just remove this check, I guess?

Reply:
Maybe just leave it the way it is for now. But my thinking is that if the tx root is correct and matches what the PoA node signed in the header, then it's likely a logical error in the client such as being out of date etc

Reply:
Oh. Well. I already removed it.